lib/httpInterface.js

```diff
@@ -268,28 +269,29 @@ function handleGetListLookup(request, reply) {
   reply.raw.setHeader('Content-Type', replyType)
   const validate = getAjvResponseValidationFunction(request.routeSchema.response['200'].items)
 
-  this.crudService
-    .aggregate(crudContext, mongoQuery, projection, sort, skip, limit, stateArr, isTextSearchQuery)
-    .stream()
-    .on('error', (error) => {
-      log.error({ error }, 'Error during findAll stream')
-      // NOTE: error from Mongo may not serialize the message, so we force it here,
-      // we use debug level to prevent leaking potetially sensible information in logs.
-      log.debug({ error: { ...error, message: error.message } }, 'Error during findAll stream with message')
-
-      if (error.code === OPTIONS_INCOMPATIBILITY_ERROR_CODE) {
-        reply.getHttpError(BAD_REQUEST_ERROR_STATUS_CODE, error.message)
-        return
-      }
-      reply.getHttpError(INTERNAL_SERVER_ERROR_STATUS_CODE, error.message || 'something went wrong')
-    })
-    .pipe(this.castResultsAsStream())
-    .pipe(validateStream(validate))
-    .pipe(serializer())
-    .pipe(reply.raw)
+
+  try {
+    return await pipeline(
+      this.crudService
+        .aggregate(crudContext, mongoQuery, projection, sort, skip, limit, stateArr, isTextSearchQuery)
+        .stream(),
+      this.castResultsAsStream(),
+      validateStream(validate),
+      serializer(),
+      reply.raw
+    )
+  } catch (error) {
+    request.log.error({ error }, 'Error during findAll lookup stream')
```

Comment:
We need to consider whether this is working properly or not. In fact, the default pino Error serializer is configured with `err` property and not error 🤔

Maybe we can bypass it by just setting the error as the object to be logged:
```suggestion
    request.log.error(error, 'Error during findAll lookup stream')
```

Reply:
Errors are already verbose, they are logged as expected, so I do not think it is necessary to edit them.